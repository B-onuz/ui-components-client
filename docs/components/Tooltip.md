# Tooltip (Component)

Componente Tooltip

## Como utilizar

```js
import Tooltip from 'ui-components/lib/components/Tooltip'
```

```jsx
import React from 'react'
import Tooltip from './Tooltip'
import { Flex, Box } from 'reflexbox/styled-components'

export default {
  title: 'Tooltip',
  component: Tooltip,
}

const tooltipShortText = 'Exemplo de Texto curto'

export const TooltipExample = () => (
  <Tooltip m={3} text={tooltipShortText}>
    Example
  </Tooltip>
)
```

## Variações

### Color

```jsx
<Tooltip color={color} />
```

### Direction

```jsx
<Tooltip direction={direction} />
```

Escala no tema: `theme.variantScale`

## Styled-system props

- space https://styled-system.com/table#space
