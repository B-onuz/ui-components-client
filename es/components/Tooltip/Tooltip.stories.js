'use strict'

Object.defineProperty(exports, '__esModule', {
  value: true,
})
exports.TooltipDirectionsExample = exports.TooltipExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Tooltip = _interopRequireDefault(require('./Tooltip'))

var _styledComponents = require('reflexbox/styled-components')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Tooltip/Tooltip.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Tooltip',
  component: _Tooltip['default'],
}
exports['default'] = _default
var tooltipLongText =
  'Exemplo de texto longo para ser exibido na ToolTip caso o componente seja utilizado para dar uma descrição maior de uma determinada funcionalidade'
var tooltipShortText = 'Exemplo de Texto curto'

var TooltipExample = function TooltipExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _styledComponents.Flex,
    {
      style: {
        height: '100vh',
      },
      width: [1],
      justifyContent: 'space-evenly',
      alignItems: 'center',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 15,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 16,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          text: tooltipShortText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 17,
            columnNumber: 7,
          },
        },
        'Example'
      )
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 21,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          text: tooltipLongText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 22,
            columnNumber: 7,
          },
        },
        'Example'
      )
    )
  )
}

exports.TooltipExample = TooltipExample

var TooltipDirectionsExample = function TooltipDirectionsExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _styledComponents.Flex,
    {
      style: {
        height: '100vh',
      },
      width: [1],
      justifyContent: 'space-evenly',
      alignItems: 'center',
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 30,
        columnNumber: 3,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 31,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          direction: 'bottom',
          color: 'violet',
          text: tooltipShortText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 32,
            columnNumber: 7,
          },
        },
        'Example'
      )
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 36,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          direction: 'top',
          color: 'olive',
          text: tooltipLongText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 37,
            columnNumber: 7,
          },
        },
        'Example'
      )
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 41,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          direction: 'right',
          text: tooltipShortText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 42,
            columnNumber: 7,
          },
        },
        'Example'
      )
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _styledComponents.Box,
      {
        p: 4,
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 46,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _Tooltip['default'],
        {
          m: 3,
          direction: 'left',
          color: 'secondary',
          text: tooltipShortText,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 47,
            columnNumber: 7,
          },
        },
        'Example'
      )
    )
  )
}

exports.TooltipDirectionsExample = TooltipDirectionsExample
