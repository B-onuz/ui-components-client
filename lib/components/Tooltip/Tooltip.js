"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _variants = require("./variants");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Tooltip/Tooltip.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  width: max-content;\n  max-width: 200px;\n  position: absolute;\n  border-radius: 12px;\n  opacity: ", ";\n  z-index: ", ";\n  display: flex;\n  flex-direction: column;\n  align-items: center;\n  padding: 8px 16px;\n  background: ", " 0% 0% no-repeat padding-box;\n  box-shadow: 0px 2px 6px #2c28281c;\n  transition: all 300ms;\n  z-index: 1000;\n  ", "\n  &::after {\n    content: '';\n    width: 1em;\n    height: 1em;\n    position: absolute;\n    box-shadow: 1px 2px 2px -1px #2c28281c;\n    background: ", " 0% 0%;\n    ", "\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose([""]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  position: relative;\n  display: flex;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledTooltipWrapper = _styledComponents["default"].div(_templateObject(), _styledSystem.space);

var ChildrenTextWrapper = _styledComponents["default"].div(_templateObject2());

var TooltipTextBox = _styledComponents["default"].div(_templateObject3(), _variants.color, function (_ref) {
  var showTooltip = _ref.showTooltip;
  return showTooltip ? 1 : 0;
}, function (_ref2) {
  var showTooltip = _ref2.showTooltip;
  return showTooltip ? 1 : -100;
}, function (_ref3) {
  var theme = _ref3.theme,
      color = _ref3.color;
  return "" + theme.colors[color];
}, function (_ref4) {
  var direction = _ref4.direction;

  if (direction === 'left') {
    return "\n      right: calc(100% + 24px);\n      top: -50%;\n    ";
  }

  if (direction === 'right') {
    return "\n      left: calc(100% + 24px);\n      top: -50%;\n    ";
  }

  if (direction === 'top') {
    return "\n      left: 50%;\n      transform: translateX(-50%);\n      bottom: calc(100% + 1.5em);\n    ";
  }

  if (direction === 'bottom') {
    return "\n      transform: translateX(-50%);\n      left: 50%;\n      top: calc(100% + 1.5em);\n    ";
  }
}, function (_ref5) {
  var theme = _ref5.theme,
      color = _ref5.color;
  return "" + theme.colors[color];
}, function (_ref6) {
  var direction = _ref6.direction;

  if (direction === 'left') {
    return "\n      right: -0.2em;\n      top: 0.6em;\n      transform: rotateZ(-45deg);\n    ";
  }

  if (direction === 'right') {
    return "\n      left: -0.2em;\n      top: 0.6em;\n      transform: rotateZ(135deg);\n    ";
  }

  if (direction === 'top') {
    return "\n      top: calc(100% - 0.5em);\n      transform: rotateZ(45deg);\n      left: 0;\n      right: 0;\n      margin: auto;\n    ";
  }

  if (direction === 'bottom') {
    return "\n      bottom: calc(100% - 0.5em);\n      transform: rotateZ(-135deg);\n      left: 0;\n      right: 0;\n      margin: auto;\n    ";
  }
});

var Tooltip = function Tooltip(_ref7) {
  var children = _ref7.children,
      direction = _ref7.direction,
      color = _ref7.color,
      text = _ref7.text,
      rest = _objectWithoutPropertiesLoose(_ref7, ["children", "direction", "color", "text"]);

  var _useState = (0, _react.useState)(false),
      showTooltip = _useState[0],
      setShowTooltip = _useState[1];

  return /*#__PURE__*/_react["default"].createElement(StyledTooltipWrapper, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 106,
      columnNumber: 5
    }
  }), /*#__PURE__*/_react["default"].createElement(ChildrenTextWrapper, {
    onMouseEnter: function onMouseEnter() {
      return setShowTooltip(true);
    },
    onMouseLeave: function onMouseLeave() {
      return setShowTooltip(false);
    },
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 107,
      columnNumber: 7
    }
  }, children), /*#__PURE__*/_react["default"].createElement(TooltipTextBox, {
    color: color,
    direction: direction,
    showTooltip: showTooltip,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 110,
      columnNumber: 7
    }
  }, text));
};

Tooltip.defaultProps = {
  color: 'primary',
  direction: 'top'
};
var _default = Tooltip;
exports["default"] = _default;
module.exports = exports.default;