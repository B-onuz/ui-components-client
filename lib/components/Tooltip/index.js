"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _Tooltip = _interopRequireDefault(require("./Tooltip"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _Tooltip["default"];
exports["default"] = _default;
module.exports = exports.default;