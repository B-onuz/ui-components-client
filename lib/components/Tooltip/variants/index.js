"use strict";

exports.__esModule = true;

var _color = _interopRequireDefault(require("./color"));

exports.color = _color["default"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }