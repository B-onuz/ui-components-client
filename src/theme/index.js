const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
