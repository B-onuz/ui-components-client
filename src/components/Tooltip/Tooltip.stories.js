import React from 'react'
import Tooltip from './Tooltip'
import { Flex, Box } from 'reflexbox/styled-components'

export default {
  title: 'Tooltip',
  component: Tooltip,
}

const tooltipLongText =
  'Exemplo de texto longo para ser exibido na ToolTip caso o componente seja utilizado para dar uma descrição maior de uma determinada funcionalidade'
const tooltipShortText = 'Exemplo de Texto curto'

export const TooltipExample = () => (
  <Flex style={{ height: '100vh' }} width={[1]} justifyContent={'space-evenly'} alignItems={'center'}>
    <Box p={4}>
      <Tooltip m={3} text={tooltipShortText}>
        Example
      </Tooltip>
    </Box>
    <Box p={4}>
      <Tooltip m={3} text={tooltipLongText}>
        Example
      </Tooltip>
    </Box>
  </Flex>
)

export const TooltipDirectionsExample = () => (
  <Flex style={{ height: '100vh' }} width={[1]} justifyContent={'space-evenly'} alignItems={'center'}>
    <Box p={4}>
      <Tooltip m={3} direction={'bottom'} color={'violet'} text={tooltipShortText}>
        Example
      </Tooltip>
    </Box>
    <Box p={4}>
      <Tooltip m={3} direction={'top'} color={'olive'} text={tooltipLongText}>
        Example
      </Tooltip>
    </Box>
    <Box p={4}>
      <Tooltip m={3} direction={'right'} text={tooltipShortText}>
        Example
      </Tooltip>
    </Box>
    <Box p={4}>
      <Tooltip m={3} direction={'left'} color={'secondary'} text={tooltipShortText}>
        Example
      </Tooltip>
    </Box>
  </Flex>
)
