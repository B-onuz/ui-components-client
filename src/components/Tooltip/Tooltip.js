import React, { useState } from 'react'
import styled from 'styled-components'
import { space } from 'styled-system'
import { color } from './variants'

const StyledTooltipWrapper = styled.div`
  ${space}
  position: relative;
  display: flex;
`

const ChildrenTextWrapper = styled.div``

const TooltipTextBox = styled.div`
  ${color}
  width: max-content;
  max-width: 200px;
  position: absolute;
  border-radius: 12px;
  opacity: ${({ showTooltip }) => (showTooltip ? 1 : 0)};
  z-index: ${({ showTooltip }) => (showTooltip ? 1 : -100)};
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px 16px;
  background: ${({ theme, color }) => `${theme.colors[color]}`} 0% 0% no-repeat padding-box;
  box-shadow: 0px 2px 6px #2c28281c;
  transition: all 300ms;
  z-index: 1000;
  ${({ direction }) => {
    if (direction === 'left') {
      return `
      right: calc(100% + 24px);
      top: -50%;
    `
    }
    if (direction === 'right') {
      return `
      left: calc(100% + 24px);
      top: -50%;
    `
    }
    if (direction === 'top') {
      return `
      left: 50%;
      transform: translateX(-50%);
      bottom: calc(100% + 1.5em);
    `
    }
    if (direction === 'bottom') {
      return `
      transform: translateX(-50%);
      left: 50%;
      top: calc(100% + 1.5em);
    `
    }
  }}
  &::after {
    content: '';
    width: 1em;
    height: 1em;
    position: absolute;
    box-shadow: 1px 2px 2px -1px #2c28281c;
    background: ${({ theme, color }) => `${theme.colors[color]}`} 0% 0%;
    ${({ direction }) => {
      if (direction === 'left') {
        return `
      right: -0.2em;
      top: 0.6em;
      transform: rotateZ(-45deg);
    `
      }
      if (direction === 'right') {
        return `
      left: -0.2em;
      top: 0.6em;
      transform: rotateZ(135deg);
    `
      }
      if (direction === 'top') {
        return `
      top: calc(100% - 0.5em);
      transform: rotateZ(45deg);
      left: 0;
      right: 0;
      margin: auto;
    `
      }
      if (direction === 'bottom') {
        return `
      bottom: calc(100% - 0.5em);
      transform: rotateZ(-135deg);
      left: 0;
      right: 0;
      margin: auto;
    `
      }
    }}
  }
`

const Tooltip = ({ children, direction, color, text, ...rest }) => {
  const [showTooltip, setShowTooltip] = useState(false)

  return (
    <StyledTooltipWrapper {...rest}>
      <ChildrenTextWrapper onMouseEnter={() => setShowTooltip(true)} onMouseLeave={() => setShowTooltip(false)}>
        {children}
      </ChildrenTextWrapper>
      <TooltipTextBox color={color} direction={direction} showTooltip={showTooltip}>
        {text}
      </TooltipTextBox>
    </StyledTooltipWrapper>
  )
}

Tooltip.defaultProps = {
  color: 'primary',
  direction: 'top',
}

export default Tooltip
