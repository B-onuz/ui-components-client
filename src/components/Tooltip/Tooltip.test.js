import React from 'react'
import Tooltip from './Tooltip.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Should render Tooltip', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Tooltip text={'Texto'}>Example</Tooltip>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
