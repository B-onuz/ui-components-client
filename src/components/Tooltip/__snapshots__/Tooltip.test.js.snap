// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Tooltip 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  color: #fff;
  background-color: #ef4d5a;
  border-color: #ef4d5a;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  max-width: 200px;
  position: absolute;
  border-radius: 12px;
  opacity: 0;
  z-index: -100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 16px;
  background: #ef4d5a 0% 0% no-repeat padding-box;
  box-shadow: 0px 2px 6px #2c28281c;
  -webkit-transition: all 300ms;
  transition: all 300ms;
  z-index: 1000;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  bottom: calc(100% + 1.5em);
}

.c1::after {
  content: '';
  width: 1em;
  height: 1em;
  position: absolute;
  box-shadow: 1px 2px 2px -1px #2c28281c;
  background: #ef4d5a 0% 0%;
  top: calc(100% - 0.5em);
  -webkit-transform: rotateZ(45deg);
  -ms-transform: rotateZ(45deg);
  transform: rotateZ(45deg);
  left: 0;
  right: 0;
  margin: auto;
}

<div
  className="c0"
>
  <div
    className=""
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    Example
  </div>
  <div
    className="c1"
    color="primary"
    direction="top"
  >
    Texto
  </div>
</div>
`;
